consensus/beacon/consensus.go

```diff
@@ -360,12 +360,6 @@ func (beacon *Beacon) Finalize(chain consensus.ChainHeaderReader, header *types.
 		amount := new(uint256.Int).SetUint64(w.Amount)
 		amount = amount.Mul(amount, uint256.NewInt(params.GWei))
 		state.AddBalance(w.Address, amount, tracing.BalanceIncreaseWithdrawal)
-
-		// Add the balance of each withdrawal to the witness, no gas will
-		// be charged.
-		if chain.Config().IsEIP4762(header.Number, header.Time) {
-			state.AccessEvents().BalanceGas(w.Address[:], true)
-		}
```

Comment:
This is removed because `AccessEvent` is only used for gas accounting. The full witness construction could potentially be handled in another structure. 

Let's just focus the gas accounting for now.

Reply:
My PR is also used to simplify a rebase. While I get this is just gas accounting from this PR's point of view, it is a layer that does build the witness.

Reply:
Myeah, my gut feeling is that witness should be something different. Basically a list of flat states needed for state execution along with their proofs, and also the input for post-state hashing.

AccessEvent just tracks which part of trie is touched.

I think accessEvent can be used as an input for witness construction, but not the witness itself.

Reply:
I agree it _will_ be something different, but I need to be able to keep rebasing as painlessly as possible until the final shape of things is built.